napalm/eos/eos.py

```diff
@@ -179,10 +195,14 @@ def _load_config(self, filename=None, config=None, replace=True):
             line = line.strip()
             if line == '':
                 continue
-            if line.startswith('!'):
+            if line.startswith('!') and not line.startswith('!!'):
                 continue
             commands.append(line)
 
+        while next((l for l in commands if isinstance(l, py23_compat.string_types)
+                    and re.match(r"(\s+)?!!", l)), None):
+            commands = self._mode_comment_convert(commands)
+
```

Comment:
Strange...it looks like I lost some earlier review comments. @bewing Can you explain what you are doing here? 

I didn't understand why you could just do an re.search and look for any `!!` lines. Also I didn't see why you needed to check if it py23_compat.string_types?

Reply:
I changed the lines you commented on, so you'd have to click the "show outdated" expander in the discussion.

Need to check for string type because I'm actively changing strings to dictionaries.

So while there continue to be strings starting with "!!" in the array, convert the next one.

Reply:
Regarding exactly what is happening:
`next()` searches for the first entry in the `commands` list that
* is a string (because re.match will die if we hand it a non-string type)
* has the regex match for `r"(\s)?!!"`

If none are found, `next()` returns None instead of raising `StopIteration`, breaking out of the while loop